File: js/GettingStarted/examples/s3client.js
var AWS = require('aws-sdk');
var S3  = require('aws-sdk/clients/s3');

AWS.config.update({
	region: 'ru-msk',
	endpoint: 'http://hb.bizmrg.com',
});

var s3 = new AWS.S3();
var bucket = "bucketbucket"
var object = "objectobject"
var body = "content"


function GetObject(BucketName, ObjectName) {
	s3.getObject({
		Bucket: BucketName,
	 	Key: ObjectName,
	},function (err, data) {
		if (err) {
	      return alert('There was an error downloading your file: ', err.message);
	    }
	    console.log('Successfully downloaded file with etag: %s', data["ETag"]);
	});
}

function PutObject(BucketName, ObjectName, Content) {
	s3.upload({
		Bucket: BucketName,
	 	Key: ObjectName,
	    Body: Content,
	    ACL: 'public-read'
	}, function(err, data) {
	    if (err) {
	      return alert('There was an error uploading your file: ', err.message);
	    }
	    console.log('Successfully uploaded file with etag: %s', data["ETag"]);
	    GetObject(BucketName, ObjectName);
	});
}


s3.createBucket({
    Bucket: bucket,
}, function(err, data){
    if (err) {
        return alert('There was an error creating bucket: ', err.message);
    }
    console.log('Successfully created bucket');
    PutObject(bucket, object, body);
})




	

File: ruby/GettingStarted/examples/s3client.rb
require 'aws-sdk'
require 'pp'

# add access_key and secret_key
Aws.config.update({
  credentials: Aws::Credentials.new('AccessKeyEXAMPLE', 'TheVeryLongLongLongSecretKeyEXAMPLE')
})

# add bucket_name, object_name and content
bucket  = 'example'
key     = 'ruby_file'
content = "content of ruby file"


s3Client = Aws::S3::Client.new(
	endpoint: 'http://hb.bizmrg.com'
)
puts "create bucket \"bucket\"\n"
s3Client.create_bucket(bucket: bucket)

puts "try to upload  file: \"#{key}\" from bucket: \"#{bucket}\" with content: \"#{content}\"\n"

put_resp = s3Client.put_object(bucket: bucket, key: key, body: content)
puts "Uploaded object with etag: "+ put_resp['etag']+"\n"

puts "try to download  file: \"#{key}\" from bucket: \"#{bucket}\" \n"
puts "Content of downloaded file:"
get_resp = s3Client.get_object(bucket: bucket, key: key) do |chunk|
     pp chunk
end
puts "Downloaded object with etag: "+ get_resp['etag']
                


